spec/unit/application/filebucket_spec.rb

```diff
@@ -114,21 +114,41 @@
         @filebucket.setup
       end
 
-      it "should default to the first server_list entry if set" do
+      it "should default to the first good server_list entry if server_list is set" do
+        stub_request(:get, "https://foo:8140/status/v1/simple/master").to_return(status: 200)
         Puppet[:server_list] = "foo,bar,baz"
         expect(Puppet::FileBucket::Dipper).to receive(:new).with(hash_including(Server: "foo"))
         @filebucket.setup
       end
 
+      it "should walk server_list until it finds a good entry" do
+        stub_request(:get, "https://foo:8140/status/v1/simple/master").to_return(status: 502)
+        stub_request(:get, "https://bar:8140/status/v1/simple/master").to_return(status: 200)
+        Puppet[:server_list] = "foo,bar,baz"
+        expect(Puppet::FileBucket::Dipper).to receive(:new).with(hash_including(Server: "bar"))
+        @filebucket.setup
+      end
+
+      # FileBucket catches any exceptions raised, logs them, then just exits
+      it "raises an error if there are no functional servers in server_list" do
+        stub_request(:get, "https://foo:8140/status/v1/simple/master").to_return(status: 404)
+        stub_request(:get, "https://bar:8140/status/v1/simple/master").to_return(status: 404)
+        Puppet[:server] = 'horacio'
+        Puppet[:server_list] = "foo,bar"
+
+        expect{@filebucket.setup}.to raise_error(SystemExit)
```

Comment:
recommend using `expect { .. }.to exit_with(0)` (or whatever the right exit code is)

Reply:
done